deepmerge/strategy/list.py

```diff
@@ -22,3 +22,8 @@ def strategy_prepend(config, path, base, nxt):
     def strategy_append(config, path, base, nxt):
         """ append nxt to base. """
         return base + nxt
+
+    @staticmethod
+    def strategy_append_unduplicated(config, path, base, nxt):
+        """ append unduplicated items in nxt to base. """
+        return base + [n for n in nxt if n not in base]
```

Comment:
```suggestion
        base_as_set = set(base)
        return base + [n for n in nxt if n not in base_as_set]
```
This may happen under the hood, but doing an "in" operator in a list is an O(n) lookup, which will get really expensive. Your idea to cast base to a set to determine the existence of an element helped reduce that operation.

looking up "in" in a set changes complexity of this from O(len(nxt) * len(base)) to O(len(nxt) + len(base)). So a polynomial to linear.